app/ide-desktop/lib/dashboard/src/authentication/src/authentication/listen.tsx

```diff
@@ -0,0 +1,66 @@
+/** @file Module for listening to authentication events emitted by Amplify.
+ *
+ * Listening to authentication events is necessary to update the authentication state of the
+ * application. For example, if the user signs out, we want to clear the authentication state so
+ * that the login screen is rendered. */
+import * as amplify from "@aws-amplify/core";
+
+// =================
+// === Constants ===
+// =================
+
+/** Name of the string identifying the "hub" that AWS Amplify issues authentication events on. */
+const AUTHENTICATION_HUB = "auth";
+
+// =================
+// === AuthEvent ===
+// =================
+
+/** Authentication state change events.
+ *
+ * These are issues by AWS Amplify when it detects a change in authentication state. For example,
+ * when the user signs in or signs out by accessing a page like `enso://auth?code=...&state=...`. */
+export enum AuthEvent {
+  /** Issued when the user has passed custom OAuth state parameters to some other auth event. */
+  customOAuthState = "customOAuthState",
+  /** Issued when the user completes the sign-in process (via federated identity provider). */
+  cognitoHostedUi = "cognitoHostedUI",
+  /** Issued when the user completes the sign-in process (via email/password). */
+  signIn = "signIn",
+}
+
+/** Returns `true` if the given `string` is an {@link AuthEvent}. */
+function isAuthEvent(value: string): value is AuthEvent {
+  return Object.values(AuthEvent).includes(value as AuthEvent);
```

Comment:
prefer this to avoid type assertion (new lint disallows type assertions)
```ts
function isAuthEvent(value: string): value is AuthEvent {
  return Object.values<string>(AuthEvent).includes(value);
}
```